In [1]:
import numpy as np

ModuleNotFoundError: No module named 'sklearn'

Среднее время сборки изделия было 90 мин. Инженер изобрел новый метод сборки этого
изделия. В результате времена сборки 10 изделий новым способом составили: 79, 74, 112,
95, 83, 96, 77, 84, 70, 90 мин. Можно ли утверждать, что среднее время сборки сократилось
(на уровне значимости 5%)?

In [29]:
data = np.array([79, 74, 112, 95, 83, 96, 77, 84, 70, 90])
m = 90

$C = \tau_{1-\varepsilon/2} = 2.262$ при $\varepsilon = 0.05$ и n - 1 = 9

$H_1$ - среднее время сборки не сократилось, $H_2$ - обратная гипотеза

Функция отклонения
$$ \rho(\vec{X}) = \sqrt{n} \frac{\overline{X} - a_0}{S_0}, S_0^2 = \frac{1}{n - 1} \sum_{i=1}^n{(X_i - \overline{X})^2}$$
$$ \delta(\vec{X}) = \begin{cases}
H_1, если |\rho(\vec{X})| < 2.262 \\
H_2, если |\rho(\vec{X})| \geq 2.262
\end{cases} $$

In [32]:
s = np.sqrt((1 / 9) * ((data - data.mean()) ** 2).sum())
print(abs(np.sqrt(9) * (data.mean() - m) / s))

0.9566892062149212


0.9566892062149212 меньше выбранного квантиля, значит можно утверждать что среднее время сборки не сократилось